frontend/src/components/ingress/Details.tsx

```diff
@@ -42,6 +43,110 @@ export default function IngressDetails() {
     );
   }
 
+  /** this function is used to determine the protocol used in the formatURL for LinkedStringFormat */
+  function getProtocol(item: any, url: any) {
+    const hostList: string[] = item.jsonData.spec?.tls?.map(({ ...hosts }) => `${hosts.hosts}`);
+    const isHttps = hostList.includes(`${url}`);
+
+    return isHttps;
+  }
+
+  /** this function is used to format the url to be used in the Link component */
+  function LinkStringFormat(url: any, item?: any, urlPath?: any) {
```

Comment:
The item is always used, as far as I can see, so it shouldn't be optional.
And the types should be precise.